### PR TITLE
improve env backup and restore

### PR DESCRIPTION
**Pull request by george43g**

### CODE REVIEW

File: libs/firebase-utils/src/lib/commands.ts [hunk 1]
```diff
@@ -104,10 +105,26 @@ export function runShellCommandExec(cmd: string) {
...
-  return runShellCommandExec(cmd);
+  return run(cmd);
+}
+
+export async function keepAlive<T>(promise: Promise<T>) {
+  let running = true;
+  async function timer(s: number) {
+    while (running) {
+      console.log('Process is running quietly...')
+      await sleep(1000 * s)
```
Comment (phpgeekfr) [inline libs/firebase-utils/src/lib/commands.ts:124]:
```suggestion
      console.log('Process is running quietly...');
      await sleep(1000 * s);
```

File: apps/backend-ops/src/backup.ts [hunk 2]
```diff
@@ -65,14 +76,18 @@ export async function restoreEnv(dirName?: string) {
   console.log('Enabling maintenance in backup...');
   await importFirestoreEmulatorBackup(firestoreURL, defaultEmulatorBackupPath);
   await enableMaintenanceInEmulator({ importFrom: 'defaultImport' });
-  await uploadDbBackupToBucket({ bucketName: backupBucket, remoteDir: `${backupDir}firestore` });
+  try {
+    await uploadDbBackupToBucket({ bucketName: backupBucket, remoteDir: `${backupDir}firestore` });
+  } catch (e) {
+    console.warn(e)
+  }
```
Comment (phpgeekfr) [inline apps/backend-ops/src/backup.ts:83]:
```suggestion
  } catch (e) console.warn(e) ;
```

File: libs/firebase-utils/src/lib/commands.ts [hunk 3]
```diff
@@ -104,10 +105,26 @@ export function runShellCommandExec(cmd: string) {
...
     cmd = `gsutil -m ${quiet ? '-q ' : ''}rm -r "${to}"`;
-    await runShellCommandExec(cmd);
+    await run(cmd);
   }
   cmd = `gsutil -m ${quiet ? '-q ' : ''}${rsync ? 'rsync' : 'cp'} ${rsync && mirror ? '-d ' : ''}${exclude ? `-x "${exclude}" ` : ''}-r "${from}" "${to}"`;
-  return runShellCommandExec(cmd);
+  return run(cmd);
+}
+
+export async function keepAlive<T>(promise: Promise<T>) {
```
Comment (phpgeekfr) [inline libs/firebase-utils/src/lib/commands.ts:119]: `export` is not needed

File: libs/firebase-utils/src/lib/commands.ts [hunk 4]
```diff
@@ -104,10 +105,26 @@ export function runShellCommandExec(cmd: string) {
...
+  let running = true;
+  async function timer(s: number) {
+    while (running) {
+      console.log('Process is running quietly...')
+      await sleep(1000 * s)
+    }
+  }
+  await Promise.race([promise, timer(60)])
```
Comment (phpgeekfr) [inline libs/firebase-utils/src/lib/commands.ts:127]:
```suggestion
  await Promise.race([promise, timer(60)]);
```

File: apps/backend-ops/src/backup.ts (hunk 2 above)
Comment (phpgeekfr) [reply apps/backend-ops/src/backup.ts]: My bad ^^ Looks like this is not working

File: libs/firebase-utils/src/lib/commands.ts (hunk 3 above)
Comment (george43g) [reply libs/firebase-utils/src/lib/commands.ts:119]: will be neeeded

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**